app/controllers/documents_v2_controller.rb

```diff
@@ -0,0 +1,119 @@
+class DocumentsV2Controller < ApplicationController
+
+  before_filter :require_valid_access_key, :only => [:open]
+  before_filter :require_valid_read_write_access_key, :only => [:save, :patch]
+  before_filter :require_valid_shared_recordid, :only => [:copy_shared]
+
+  # don't check for csrf token
+  skip_before_filter :verify_authenticity_token, :only => [:save, :patch, :copy_shared]
+
+  # v2 of the api is all anonymous access
+  skip_authorization_check
+
+  include DocumentsV2Helper
+
+  def open
```

Comment:
Ok - I did it this way to keep it in the same style as the documents controller.  I'll inline it and return the document.